ref/general/secure-communication-tls.adoc

```diff
@@ -0,0 +1,86 @@
+In order for a website to comply with the secure HTTPS protocol, its communications must be encrypted according to the TLS protocol. Most web browsers flag sites that do not comply with the HTTPS protocol as `not secure`.
+
+== TLS protocol
+
+The TLS protocol is made up of two component protocols: the TLS handshake and the TLS record.
+The TLS handshake sets the terms for secure communication between client and server during the session.
+The TLS record secures and verifies the integrity of the data that is exchanged.
+
+TLS provides authentication, confidentiality, and integrity for internet communications.
+Authentication confirms that each communicating party is who they claim to be through the exchange of asymmetrically encrypted keys and digitally signed certificates.
+This exchange takes place before any message data is transmitted, during the TLS handshake.
+
+TLS ensures confidentiality through a combination of symmetric and asymmetric encryption.
+During the TLS handshake, the client and server agree upon an encryption algorithm and a shared secret key that are used for that session only.
+All messages that are transmitted between the client and server are encrypted by using that algorithm and key, which only the communicating parties can access.
+This configuration helps ensure that even if messages are intercepted, they remain private.
+
+TLS-secured messages establish data integrity by creating a digital signature that is used to verify that a message was not tampered with. A digital signature is formed by computing and encrypting a message digest, which is transmitted together with the full encrypted message.
+Upon receipt and decryption of the message, the receiver also computes a message digest, which is compared with the decrypted digest from the sender.
+If the two digests match, the sender knows that the message was not modified during transmission.
+
+=== TLS and SSL
+
+TLS updates and replaces SSL, which is now deprecated though still in use in some older devices.
+TLS is more secure than SSL, thanks to stronger encryption algorithms and better message authentication.
+TLS addresses many known security vulnerabilities from SSL, and adds extra capabilities, such as secure remote passwords and Kerberos support.
+Although TLS replaces SSL for most websites, the term SSL is still widely used as a colloquialism and in many cases can refer to either SSL or TLS-enabled connections.
+TLS and SSL are not interoperable, though TLS does provide compatibility support for older devices that still use SSL.
+
+=== TLS handshakes
+
+The TLS handshake is a process by which a client and server set the terms for secure communications during a connection.
+The handshake occurs before any data is exchanged.
+Through this process, the client and server agree on the version of the TLS protocol to use, select cryptographic algorithms (CypherSuites), and authenticate one another by exchanging and verifying digital certificates.
```

Comment:
"Through this process, the client and server agree on the version of the TLS protocol to use, select cryptographic algorithms (CypherSuites), and authenticate one another by exchanging and verifying digital certificates." in this sentence is there any alternative for the parentheses in (CypherSuites)?